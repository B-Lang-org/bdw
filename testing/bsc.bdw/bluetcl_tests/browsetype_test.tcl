namespace import ::Bluetcl::*

# Loads the specified package
puts [bpackage load Test]
puts ""

# Adds the specified type to the type hierarchy
puts [browsetype add [type constr Sz]]

# Shows up level of the type hierarchy
puts [browsetype list 0]

# Expands the specified node of the type hierarchy
puts [browsetype list 1]

# Adds the specified type to the type hierarchy
puts [browsetype add [type constr Bar]]

# Adds the specified type to the type hierarchy
puts [browsetype add {TopIFC#(ty)}]

# Refreshes the type hierarchy
puts [browsetype refresh]

# Shows up level of the type hierarchy
puts [browsetype list 0]

# Expands the specified node of the type hierarchy
puts [browsetype list 1]
puts [browsetype list 2]

# Refreshes the type hierarchy
puts [browsetype clear]

# Refreshes the type hierarchy
puts [browsetype refresh]

# Shows up level of the type hierarchy
puts [browsetype list 0]
