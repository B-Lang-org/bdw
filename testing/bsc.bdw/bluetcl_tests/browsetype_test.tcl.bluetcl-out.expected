Test Vector Array List PreludeBSV Prelude


{1 {Sz (alias)} {}}
{2 {9 (numeric type)} {}}



{1 {Sz (alias)} {}} {2 {Bar (enum) (2)} {}} {3 {TopIFC#(ty) (interface)} {}}
{4 {9 (numeric type)} {}}




